app/notification_push.go

```diff
@@ -316,7 +316,7 @@ func DoesNotifyPropsAllowPushNotification(user *model.User, channelNotifyProps m
 	channelNotify, ok := channelNotifyProps[model.PUSH_NOTIFY_PROP]
 
 	// If the channel is muted do not send push notifications
-	if channelMuted, ok := channelNotifyProps[model.MARK_UNREAD_NOTIFY_PROP]; ok {
```

Comment:
This `ok` (now `channelMutedOk`) and the `channelMuted` variables appear to be redundant: the `nil` value of the lookup will be the empty string, and trivially fail the check immediately thereafter. Propose the following instead:
```
if channelNotifyProps[model.MARK_UNREAD_NOTIFY_PROP] == model.CHANNEL_MARK_UNREAD_MENTION {
    return false
}